cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/output/MultipleOutputsMainOutputWrapper.java

```diff
@@ -107,24 +115,27 @@ public synchronized OutputCommitter getOutputCommitter(TaskAttemptContext contex
       Map<String, OutputCommitter> delegates = new LinkedHashMap<>();
 
       List<String> namedOutputsList = MultipleOutputs.getNamedOutputsList(context);
+      TaskAttemptContext namedContext = context;
       // if there is only 1 output configured, it is the same as the root OutputFormat, so no need to have it also in
       // the delegates; otherwise, its methods would get called more than expected.
       // If more than 1 outputs are configured, then the root OutputCommitter is a NullOutputCommitter (no-op).
       // See MapReduceRuntimeService#setOutputsIfNeeded.
-      if (namedOutputsList.size() > 1) {
-        for (String name : namedOutputsList) {
+      for (String name : namedOutputsList) {
+        // CDAP-14524 Get context without prefixes for single and multiple outputs.
+        namedContext = MultipleOutputs.getNamedTaskContext(context, name);
+        if (namedOutputsList.size() > 1) {
```

Comment:
doesn't this change the behavior so that the root is now added to the delegates? Why does it need to be in delegates and also passed in a separate variable? Can't that cause problems if it tries to commit twice?

Reply:
The functionality is still the same, will modify the code structure to make it more readable